### DISCUSSION TITLE
Connection reset on SignServer AWS

### DISCUSSION THREAD

**Discussion by LVIncoder** [Keyfactor/signserver-ce#37] [Q&A] [answered]
I launch a new Signserver instance in AWS following the quick guide on the documentation site, and the instance seems to be running without a problem, but when I try to go to the keystore I receive an error that says "The connection was reset". 

When I go to the instance log I got this

dist:zip:
      [zip] Building zip: /opt/signserver/tmp/signserver-clientcli.zip
     [echo] Wrote /opt/signserver/tmp/signserver-clientcli.zip

-post-configure-ear:

configure-ear:

-pre-deploy-ear:

-set-deploy-dir-jboss:

j2ee:configure:

j2ee:deploy:
     [echo] Deploying /opt/signserver-ee-5.11.1.Final/tmp/signserver.ear to /opt/wildfly/standalone/deployments
     [copy] Copying 1 file to /opt/wildfly/standalone/deployments
     [echo]

showtime:
     [echo] Task completed 2023-02-23 00:39:01 +0000.

-do-deploy-ear:

-post-deploy-ear:

deploy-ear:

deploy:

BUILD SUCCESSFUL
Total time: 27 seconds
adding: info(in = 505) (out= 293)(deflated 41%)
** Starting SignServer service via WildFly...
** CA service (Application Server) not running.
Waiting for WildFly to start SignServer...
Waiting for WildFly to start SignServer...
Waiting for WildFly to start SignServer...
Waiting for WildFly to start SignServer...
Waiting for WildFly to start SignServer...

Am I missing something in order to get the Signserver deploy correctly?

**Comment by agregory23** [comment]
Hi @LVIncoder-

Thanks for submitting this.  It looks like a parsing bug got introduced into the latest version.  You can fix this by first stopping wildfly and then going to the /opt/wildfly/bin/standalone.conf file and change:

`   JAVA_OPTS="-Xms2048 -Xmx2048mXMx}m -Xss256k -XX:MetaspaceSize=256M -XX:MaxMetaspaceSize=${jvmXXMaxMetaspaceSize}m -Djava.net.preferIPv4Stack=true -Dsun.java2d.print.polling=false"`
   
   to
   
`   JAVA_OPTS="-Xms2048m -Xmx${jvmXMx}m -Xss256k -XX:MetaspaceSize=256M -XX:MaxMetaspaceSize=${jvmXXMaxMetaspaceSize}m -Djava.net.preferIPv4Stack=true -Dsun.java2d.print.polling=false"`

Then start wildfly up again and you should see the install pick up where it left off and finish.

I will have a new version pushed to AWS soon and you can launch again as well.

**Comment by LVIncoder** [comment]
Hi @agregory23, 

Thanks for the respose, I made the changes you mention but still can´t run the instance. Now is showing this:

[ec2-user@ip-192-168-90-4 wildfly]$ sudo ./bin/standalone.sh

**Detected 1064GB available memory.  Thats less than 2GB.  This install may not work.**
=========================================================================

  JBoss Bootstrap Environment

  JBOSS_HOME: /opt/wildfly-24.0.1.Final

  JAVA: java

  JAVA_OPTS:  -server -Xms2048m -Xmx728m -Xss256k -XX:MetaspaceSize=256M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Dsun.java2d.print.polling=false -Dsun.rmi.dgc.client.gcInterval=3600000 -Dsun.rmi.dgc.server.gcInterval=3600000 -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true -Djboss.server.default.config=standalone.xml -Djboss.tx.node.id=53 -Dcos.database.address=localhost -Dcos.database.name=signserver -Dcos.database.username=signserver -Dcos.database.password=i-05b25e2c49543bb7f  --add-exports=java.desktop/sun.awt=ALL-UNNAMED --add-exports=java.naming/com.sun.jndi.ldap=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.security=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.management/javax.management=ALL-UNNAMED --add-opens=java.naming/javax.naming=ALL-UNNAMED

=========================================================================

Error occurred during initialization of VM
Initial heap size set to a larger value than the maximum heap size

**Reply by agregory23** [reply]
What size instance are you running?  You need to run at minimum a t2 or t3.medium.  WildFly does not do well with an on box database and less than 4GB of RAM for itself the OS and the database.  SignServer also needs at least 2 cores to function since its a multi threaded app.  If you shut down the instance in the AWS Console and right click, then select change instance size it should come up.

**Comment by LVIncoder** [comment]
As currently we are on a POC we have a t3a.medium. (2vCpu, 4GB RAM). We already launch a new instance and do the configuration file change, but still getting the same error

**Comment by agregory23** [comment, marked as answer]
OK.. sounds good.  Let’s just get you the new version.  I made some changes and have tested it’s working.  The new version is pushing now.  AWS is pretty quick about this.  It should be live in a few hours.

Apologies for you having any issues.  I recently changed how the instance calculates the free RAM more dynamically so we can support more instance types.  I will update when its live.  IT will be version 1.13.2.

**Comment by agregory23** [comment]
New version is live.  You should be able to terminate the old one and run a new one.

**Comment by LVIncoder** [comment]
SignServer is running now!! Thanks @agregory23

**Comment by LVIncoder** [comment]
@agregory23 we are running now SignServer and im able to get to the keystore, but when I try to download the .p12 a 403 error appears.

I download the .p12 by SCP and take it to the Firefox keyvault but im not able to access the adminweb, every time i try to go there, the service redirects to keystore
